gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/diagnosticreport/SpecimenBatteryMapperTest.java

```diff
@@ -37,144 +35,297 @@
 import uk.nhs.adaptors.pss.translator.util.DegradedCodeableConcepts;
 import static uk.nhs.adaptors.common.util.CodeableConceptUtils.createCodeableConcept;
+import static uk.nhs.adaptors.pss.translator.util.XmlUnmarshallUtil.unmarshallString;
 
 @ExtendWith(MockitoExtension.class)
 public class SpecimenBatteryMapperTest {
 
-    private static final String RESOURCES_BASE = "xml/SpecimenBattery/";
-
-    private static final String BATTERY_CLASSCODE = "BATTERY";
-    private static final String PRACTISE_CODE = "TEST_PRACTISE_CODE";
-    private static final String OBSERVATION_ID = "SPECIMEN_CHILD_BATTERY_COMPOUND_STATEMENT_ID_1";
-    private static final String OBSERVATION_STATEMENT_ID_1 = "BATTERY_DIRECT_CHILD_OBSERVATION_STATEMENT";
-    private static final String OBSERVATION_STATEMENT_ID_2 = "OBSERVATION_STATEMENT_ID";
-    private static final String DIAGNOSTIC_REPORT_ID = "DIAGNOSTIC_REPORT_ID";
-    private static final String ENCOUNTER_ID = "ENCOUNTER_ID";
-    private static final String PATIENT_ID = "TEST_PATIENT_ID";
-    private static final String SPECIMEN_ID = "TEST_SPECIMEN_ID_1";
-    private static final String META_PROFILE_SUFFIX = "Observation-1";
-    private static final String EXPECTED_COMMENT = "Looks like Covid";
-    private static final Patient PATIENT = (Patient) new Patient().setId(PATIENT_ID);
-    private static final DiagnosticReport DIAGNOSTIC_REPORT = (DiagnosticReport) new DiagnosticReport().setId(DIAGNOSTIC_REPORT_ID);
-    private static final InstantType OBSERVATION_ISSUED = parseToInstantType("202203021160700");
-    private static final DateTimeType OBSERVATION_EFFECTIVE = parseToDateTimeType("20100223000000");
-    private static final String CODING_DISPLAY_MOCK = "Test Display";
-    private static final String SNOMED_SYSTEM = "http://snomed.info/sct";
-    private static final CodeableConcept CODEABLE_CONCEPT = createCodeableConcept(null, SNOMED_SYSTEM, CODING_DISPLAY_MOCK);
-    private final List<Encounter> encounters = generateEncounters();
+    public static final String EHR_EXTRACT_WRAPPER = """
+        <EhrExtract xmlns="urn:hl7-org:v3" classCode="EXTRACT" moodCode="EVN">
+            <component>
+                <ehrFolder>
+                    <component>
+                        {{ehrComposition}}
+                    </component>
+                </ehrFolder>
+            </component>
+        </EhrExtract>
+        """;
 
     @Mock
     private CodeableConceptMapper codeableConceptMapper;
 
     @InjectMocks
     private SpecimenBatteryMapper specimenBatteryMapper;
 
-    @Test
-    public void testMappingObservationFromBatteryCompoundStatement() {
-        final RCMRMT030101UK04EhrExtract ehrExtract = unmarshallEhrExtract("specimen_battery_compound_statement.xml");
-        var batteryCompoundStatement = getBatteryCompoundStatements(ehrExtract);
+    @Test void When_MappingObservationWithEffectiveTimeInBatteryCompoundStatement_Expect_IssuedUsesThisValue() {
```

Comment:
effectiveTime or availabilityTime? I'm confused by this.

Reply:
AvailabilityTime!  Have changed the names of the methods so they refer to this instead of effective time now!